source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -589,6 +597,17 @@ inline void ComponentInterface<NodeT>::set_parameter_value(const std::string& na
   }
 }
 
+template<class NodeT>
+inline bool ComponentInterface<NodeT>::validate_parameter_wrapper(
+    const std::shared_ptr<state_representation::ParameterInterface>& parameter
+) {
+  if (parameter->get_name() == "period" && parameter->get_parameter_value<double>() < 1e-3) {
```

Comment:
I'm not sure I agree with this magic number limit. 1kHz is certainly a reasonable upper bound but it might change on specialised hardware with real-time determinism. This number should be somewhere higher up so that it can be adjusted if necessary; my suggestion would be a `minimum_period_` private property, for which we can add a protected setter.

Reply:
agree with you. I wouldn't put upper bounds to be honest. Some robots now run at 2kHz

Reply:
Actually, thinking about this more, I think it should just be invalid at `<= 0.0`. Any positive real number will work because it signifies the target periodicity - if the executor / CPU can't keep up then the real period will be slower automatically. At some point we'll add watchdog monitoring and diagnostics to the main engine so we can see the process load, in which case we can warn if a component period is too low to be achieved. I don't think it needs to be done at the component level.